src/account/operations/syncing/addresses/output_ids/mod.rs

```diff
@@ -34,72 +35,114 @@ impl AccountHandle {
             return Ok(output_ids);
         }
 
+        #[cfg(target_family = "wasm")]
+        let mut results = vec![];
+
+        #[cfg(not(target_family = "wasm"))]
         let mut tasks = vec![];
 
         if (address.is_ed25519() && sync_options.account.basic_outputs)
             || (address.is_nft() && sync_options.nft.basic_outputs)
             || (address.is_alias() && sync_options.alias.basic_outputs)
         {
             // basic outputs
-            tasks.push(
-                async move {
-                    let account_handle = self.clone();
-                    let bech32_address_ = bech32_address.clone();
-                    tokio::spawn(async move {
-                        account_handle
-                            .get_basic_output_ids_with_any_unlock_condition(&bech32_address_)
-                            .await
-                    })
-                    .await
-                }
-                .boxed(),
-            );
+            #[cfg(target_family = "wasm")]
+            {
+                results.push(
+                    self.clone()
```

Comment:
That clone is a bit weird. `get_basic_output_ids_with_any_unlock_condition` shouldn't take ownership, can you change that ?